contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -3416,9 +3416,39 @@ LANGUAGE C STABLE PARALLEL SAFE;
 CREATE OR REPLACE FUNCTION sys.is_member(IN role sys.SYSNAME)
 RETURNS INT AS
 $$
-	SELECT sys.is_rolemember_internal(role, NULL);
+BEGIN
+    -- Always return 1 for 'public'
+    IF (role = 'public')
+    THEN RETURN 1;
+    END IF;
+
+    IF EXISTS (SELECT orig_loginname FROM sys.babelfish_authid_login_ext WHERE orig_loginname = role)
+    THEN
+        IF EXISTS (SELECT name FROM sys.login_token WHERE name = role)
+        THEN RETURN 1; -- Return 1 if current session user is a member of role or windows group
+        ELSIF NOT EXISTS (SELECT type FROM sys.login_token WHERE type IN ('WINDOWS LOGIN', 'WINDOWS GROUP')) THEN -- session is not a windows auth session
+            IF (CHARINDEX('\', role) != 0)
+            THEN RETURN NULL;  -- argument is a windows group
```

Comment:
Shouldn't we return 1 here? Why NULL?

Reply:
If session is not a windows authenticated session but argument is a windows group, we should return NULL as per T-SQL specifications